### PR TITLE
add M3D-RPN model

### PR DESCRIPTION
**Pull request by shuluoshu**
**add M3D-RPN model**
https://arxiv.org/pdf/1907.06038.pdf

### CODE REVIEW

File: PaddleCV/3d_vision/M3D-RPN/lib/core.py [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+    # make directories
+    mkdir_if_missing(paths.output)
+    mkdir_if_missing(paths.logs)
+    mkdir_if_missing(paths.weights)
+    mkdir_if_missing(paths.results)
+
+    return paths
+
+
+# def init_torch(rng_seed, cuda_seed):
```
Comment (qingqing01) [inline PaddleCV/3d_vision/M3D-RPN/lib/core.py]: clear code

File: PaddleCV/3d_vision/M3D-RPN/lib/core.py [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+        tracker[id].append(val)
+
+        # store object info
+        obj_id = id + '_obj'
+        if not (obj_id in tracker):
+            stat.pop('val', None)
+            tracker[id + '_obj'] = stat
+
+
+# def next_iteration(loader, iterator):
```
Comment (qingqing01) [inline PaddleCV/3d_vision/M3D-RPN/lib/core.py]: need to clear code

**Review by qingqing01 — COMMENTED** [review]
先把代码清理下吧

File: PaddleCV/3d_vision/M3D-RPN/models/backbone/__init__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+"""
+#copyright (c) 2019 PaddlePaddle Authors. All Rights Reserve.
```
Comment (heavengate) [inline PaddleCV/3d_vision/M3D-RPN/models/backbone/__init__.py]: 2019 -> 2020

Comment (shuluoshu) [reply PaddleCV/3d_vision/M3D-RPN/models/backbone/__init__.py]: 已修正2020

File: PaddleCV/3d_vision/M3D-RPN/utils.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+
+class Stat(object):
+    """Stat"""
+
+    def __init__(self):
+        self.stats = {}
+
+    def update(self, keys, values):
+        """update"""
```
Comment (heavengate) [inline PaddleCV/3d_vision/M3D-RPN/utils.py:89]: 删掉或者细化一下？

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**